data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/CountAggregateAction.java

```diff
@@ -81,15 +79,33 @@ public Exemplar createExemplar(final Event event) {
     @Override
     public AggregateActionResponse handleEvent(final Event event, final AggregateActionInput aggregateActionInput) {
         final GroupState groupState = aggregateActionInput.getGroupState();
+        Instant eventStartTime = Instant.now();
+        Instant eventEndTime = eventStartTime;
+        Object startTime = event.get(startTimeKey, Object.class);
+        Object endTime = event.get(endTimeKey, Object.class);
+
+        if (startTime != null) {
+            eventStartTime = AggregateProcessor.convertObjectToInstant(startTime);
+        }
+        if (endTime != null) {
+            eventEndTime = AggregateProcessor.convertObjectToInstant(endTime);
+        }
         if (groupState.get(countKey) == null) {
-            groupState.put(startTimeKey, Instant.now());
             groupState.putAll(aggregateActionInput.getIdentificationKeys());
             groupState.put(countKey, 1);
             groupState.put(exemplarKey, createExemplar(event));
+            groupState.put(startTimeKey, eventStartTime);
+            groupState.put(endTimeKey, eventEndTime);
         } else {
             Integer v = (Integer)groupState.get(countKey) + 1;
             groupState.put(countKey, v);
-        } 
+            Instant groupStartTime = (Instant)groupState.get(startTimeKey);
+            Instant groupEndTime = (Instant)groupState.get(endTimeKey);
+            if (eventStartTime.isBefore(groupStartTime))
+                groupState.put(startTimeKey, eventStartTime);
+            if (eventEndTime.isAfter(groupEndTime))
+                groupState.put(endTimeKey, eventEndTime);
```

Comment:
This seems like generic aggregation functionality where we just track the range of times the events have come from for that aggregated event. Is there a way to extend this to all aggregate actions with some common code? Or is it only applicable to count and histogram actions?